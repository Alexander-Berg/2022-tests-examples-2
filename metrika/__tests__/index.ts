import { remap } from '../normalize';

describe('remap', () => {
    it('works', () => {
        const result = remap(
            { yandexEmployeeLogin: '13' },
            { 13: 'loginFromStaff' },
        );

        expect(result).toEqual({ yandexEmployeeLogin: 'loginFromStaff' });
    });

    it('skips keys if they not presented in raw object', () => {
        const result = remap({ yandexEmployeeLogin: '13' }, {});

        expect(result).toEqual({});
    });

    it('skips keys if they not presented in mapping object', () => {
        const result = remap({}, { 13: 'loginFromStaff' });

        expect(result).toEqual({});
    });

    it('mixed case', () => {
        const result = remap(
            { someOtherField: '42' },
            { 13: 'loginFromStaff' },
        );

        expect(result).toEqual({});
    });
});
