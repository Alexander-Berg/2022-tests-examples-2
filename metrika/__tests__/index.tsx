import * as React from 'react';

import { render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Highlight, Substr } from '..';
configure({ adapter: new Adapter() });

describe('Highlight', () => {
    it('работает по строке', () => {
        const container = render(
            <div>
                <Highlight pattern="екс">текст</Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с вложенными строками', () => {
        const container = render(
            <div>
                <Highlight pattern="екс">
                    <span>текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с несколькими вложенными строками', () => {
        const container = render(
            <div>
                <Highlight pattern="екс">
                    <span>текст</span>
                    <span>текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с несколькими паттернами через пробел', () => {
        const container = render(
            <div>
                <Highlight pattern="ако екс">
                    <span>какой-то текст</span>
                    <span>какой-то текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с паттернами через массив', () => {
        const container = render(
            <div>
                <Highlight pattern={['ако', 'екс']}>
                    <span>какой-то текст</span>
                    <span>какой-то текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с подсветкой фона', () => {
        const container = render(
            <div>
                <Highlight pattern={['ако', 'екс']} highlight="background">
                    <span>какой-то текст</span>
                    <span>какой-то текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает позиционная подсветка', () => {
        const container = render(
            <div>
                <Substr match={{ start: 1, end: 3 }}>Here comes Johny!</Substr>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
});
