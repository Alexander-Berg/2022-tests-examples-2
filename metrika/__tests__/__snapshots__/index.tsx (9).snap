// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`size "m", state "closed" 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  width: 18px;
  height: 18px;
  vertical-align: middle;
  text-align: center;
  line-height: 17px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 {
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  border-radius: 3px;
  box-sizing: border-box;
  border: 1px solid #b8bec3;
  vertical-align: middle;
  text-align: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 .c2 {
  color: #b8bec3;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  left: 50%;
}

.c1 .c3 {
  color: #b8bec3;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  left: 50%;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <span
      class="BaseIcon IconPlus c2 "
      data-lego="react"
    >
      <svg
        fill="currentColor"
        height="9px"
        viewBox="0 0 8 8"
        width="9px"
      >
        <path
          d="M4.696 3.335h3.333v1.333H4.696v3.334H3.363V4.668H.029V3.335h3.334V.002h1.333z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`size "m", state "loading" 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  width: 18px;
  height: 18px;
  vertical-align: middle;
  text-align: center;
  line-height: 17px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div
  class="c0"
>
  <div
    class="spin2 spin2_size_xxs spin2_view_default spin2_tone_default spin2_progress_yes"
    data-lego="react"
  />
</div>
`;

exports[`size "m", state "opened" 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  width: 18px;
  height: 18px;
  vertical-align: middle;
  text-align: center;
  line-height: 17px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 {
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  border-radius: 3px;
  box-sizing: border-box;
  border: 1px solid #b8bec3;
  vertical-align: middle;
  text-align: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 .c3 {
  color: #b8bec3;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  left: 50%;
}

.c1 .c2 {
  color: #b8bec3;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  left: 50%;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <span
      class="BaseIcon IconMinus c2 "
      data-lego="react"
    >
      <svg
        fill="currentColor"
        height="9px"
        viewBox="0 0 8 8"
        width="9px"
      >
        <path
          d="M8.03 3.335v1.333h-8V3.335z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`size "s", state "closed" 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  width: 14px;
  height: 14px;
  vertical-align: middle;
  text-align: center;
  line-height: 13px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 {
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  border-radius: 3px;
  box-sizing: border-box;
  border: 1px solid #b8bec3;
  vertical-align: middle;
  text-align: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 .c2 {
  color: #b8bec3;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  left: 50%;
}

.c1 .c3 {
  color: #b8bec3;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  left: 50%;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <span
      class="BaseIcon IconPlus c2 "
      data-lego="react"
    >
      <svg
        fill="currentColor"
        height="7px"
        viewBox="0 0 8 8"
        width="7px"
      >
        <path
          d="M4.696 3.335h3.333v1.333H4.696v3.334H3.363V4.668H.029V3.335h3.334V.002h1.333z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`size "s", state "loading" 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  width: 14px;
  height: 14px;
  vertical-align: middle;
  text-align: center;
  line-height: 13px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div
  class="c0"
>
  <div
    class="spin2 spin2_size_xxs spin2_view_default spin2_tone_default spin2_progress_yes"
    data-lego="react"
  />
</div>
`;

exports[`size "s", state "opened" 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  width: 14px;
  height: 14px;
  vertical-align: middle;
  text-align: center;
  line-height: 13px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 {
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  border-radius: 3px;
  box-sizing: border-box;
  border: 1px solid #b8bec3;
  vertical-align: middle;
  text-align: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 .c3 {
  color: #b8bec3;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  left: 50%;
}

.c1 .c2 {
  color: #b8bec3;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  left: 50%;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <span
      class="BaseIcon IconMinus c2 "
      data-lego="react"
    >
      <svg
        fill="currentColor"
        height="7px"
        viewBox="0 0 8 8"
        width="7px"
      >
        <path
          d="M8.03 3.335v1.333h-8V3.335z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </div>
</div>
`;
